torch/testing/_internal/common_methods_invocations.py

```diff
@@ -9342,6 +9347,8 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                DecorateInfo(unittest.skip("Skipped!"), 'TestGradients', 'test_fn_gradgrad'),
                # GradcheckError: gradcheck expects all tensor inputs are dense when check_sparse_nnz is set to False
                DecorateInfo(unittest.skip("Skipped!"), 'TestGradients', 'test_forward_mode_AD'),
+               # Some inputs are non deterministic, eg when mat1 and mat2 are size 0
+               DecorateInfo(unittest.skip('some inputs are non deterministic'), 'TestCommon', 'test_compare_cpu'),
```

Comment:
@amjames here is the case that is failing: `1,1,1,1,1 != 0.2,0.2,0.2,0.2,0.2`  

```py
In [19]: torch.sparse.sampled_addmm(torch.eye(5).to_sparse_csr(), torch.zeros([5,0]), torch.zeros([0,5]), alpha=0.6, beta=0.2)
Out[19]: 
tensor(crow_indices=tensor([0, 1, 2, 3, 4, 5]),
       col_indices=tensor([0, 1, 2, 3, 4]),
       values=tensor([0.2000, 0.2000, 0.2000, 0.2000, 0.2000]), size=(5, 5),
       nnz=5, layout=torch.sparse_csr)

In [20]: torch.sparse.sampled_addmm(torch.eye(5, device='cuda:0').to_sparse_csr(), torch.zeros([5,0], device='cuda:0'), torch.zeros([0,5], device='cuda:0'), alpha=0.6, beta=0.2)
Out[20]: 
tensor(crow_indices=tensor([0, 1, 2, 3, 4, 5]),
       col_indices=tensor([0, 1, 2, 3, 4]),
       values=tensor([1., 1., 1., 1., 1.]), device='cuda:0', size=(5, 5),
       nnz=5, layout=torch.sparse_csr)
```

Reply:
This is a real bug in the cuda implementation. You should open an issue.